source/AI.cpp

```diff
@@ -1978,7 +1981,21 @@ void AI::PrepareForHyperspace(Ship &ship, Command &command)
 	bool isJump = (ship.GetCheapestJumpType(ship.GetTargetSystem()).first == Ship::JumpType::JumpDrive);
 
 	Point direction = ship.GetTargetSystem()->Position() - ship.GetSystem()->Position();
-	if(!isJump && scramThreshold)
+	double departure =  isJump ?
+		ship.GetSystem()->JumpDepartureDistance() :
+		ship.GetSystem()->HyperDepartureDistance();
+	double squaredDeparture = departure * departure + SAFETY_OFFSET;
+	if(ship.Position().LengthSquared() < squaredDeparture)
+	{
+		Point closestDeparturePoint;
+		if(ship.Position())
+			closestDeparturePoint = ship.Position()
+				* (squaredDeparture / ship.Position().LengthSquared());
+		else
+			closestDeparturePoint = Point(1., squaredDeparture);
+		MoveTo(ship, command, closestDeparturePoint, Point(.0, .0), .0, .0);
```

Comment:
```suggestion
		MoveTo(ship, command, closestDeparturePoint, Point(), 0., 0.);
```

~~actually the last one now~~

Reply:
...